Orange/data/table/impl.py

```diff
@@ -0,0 +1,427 @@
+            # in a nutshell, gets the coo_matrix building components directly
+            # from each column of the SparseTable
+            result_data = []
+            result_row = []
+            result_col = []
+            for i, col in enumerate(cols):
+                column_index = self[col.name].sp_index
+                if isinstance(column_index, BlockIndex):
+                    column_index = column_index.to_int_index()
+                result_data.append(self[col.name].sp_values)
+                result_row.append(column_index.indices)
+                result_col.append(len(self[col.name].sp_values) * [i])
+            return sp.coo_matrix((np.concatenate(result_data), (np.concatenate(result_row), np.concatenate(result_col))),
+                                 (n_rows, len(cols)), dtype=np.float64)
+
+    @property
+    def Y(self):
+        result = self._to_numpy(Y=True)
+        # see _to_numpy for why there are two cases
+        if sp.issparse(result):
+            # subscripting sparse matrices doesn't work, so get a copy
+            return result.getcol(0).tocoo() if result.shape[1] == 1 else result
+        else:
+            return result[:, 0] if result.shape[1] == 1 else result
+
+    @classmethod
+    def _coo_to_sparse_dataframe(cls, coo_matrix, column_index_start):
+        """Convert a scipy.sparse.coo_matrix into a sparse dataframe.
+
+        Indices start at 0.
+
+        This constructs a single SparseTableSeries, which is then used to fill up
+        the resulting SparseTable column-by-column. This is, counter-intuitively,
+        a good way of doing things.
+
+        We can't create a SparseDataFrame from a SparseSeries and then unstack
+        the MultiIndex, because that gives us a dense DataFrame.
+
+        We can't pass a list of SparseSeries rows to the SparseDataFrame constructor,
+        as we would with a dense DataFrame, because that doesn't (yet?) work.
+
+        Parameters
+        ----------
+        coo_matrix : scipy.sparse.coo_matrix
+            The origin COOrdinate matrix.
+        column_index_start : int
+            Where to start counting the columns from.
+            Useful for later concatenating multiple DataFrames.
+
+        Returns
+        -------
+        pd.SparseDataFrame
+            The resulting sparse data frame.
+            Not a SparseTable yet because we don't have any Orange-specific data associated with it.
+        """
+        # convert into a multiindex sparse series
+        # transposed because it's easier to get rows from ss
+        # use a dense index so this works when a row is all-nan
+        ss = pd.SparseSeries.from_coo(coo_matrix.T, dense_index=True)
+
+        # create a new, completely empty sparse container
+        # and fill its columns up sequentially
+        # use a SparseDataFrame so we don't do any weights
+        result = pd.SparseDataFrame(index=list(range(coo_matrix.shape[0])),
```

Comment:
This is just for transformations. The index is then constructed when all of these are joined, at the bottom of `_from_sparse_numpy`. Or did you have something else in mind?